pulse-binding/src/context/introspect.rs

```diff
@@ -452,6 +462,27 @@ impl<'a> SinkInfo<'a> {
             }
         }
     }
+
+    /// Creates owned data from borrowed data
+    pub fn to_owned(&self) -> SinkInfo<'static> {
+        SinkInfo {
+            name: self.name.clone().map(|o| Cow::Owned(o.into_owned())),
+            description: self.description.clone().map(|o| Cow::Owned(o.into_owned())),
+            monitor_source_name: self
+                .monitor_source_name
+                .clone()
+                .map(|o| Cow::Owned(o.into_owned())),
+            driver: self.driver.clone().map(|o| Cow::Owned(o.into_owned())),
+            proplist: self.proplist.clone(),
```

Comment:
This doesn't, or rather shouldn't, work as there's the potential for a use-after-free. `Proplist` contains `ProplistInner` which itself contains a raw pointer to a string array. We would need to use the iteration functionality to iterate over and clone the set of strings and collect them into a vector. Then of course we would need to modify this attribute such that it can hold either the borrowed or cloned version of the strings.

Reply:
I think that's fine actually, `Clone` for `Proplist` seems to do a deep copy
```rust
impl Clone for Proplist {
    /// Allocates a new property list and copy over every single entry from the specified list.
    ///
    /// If this is called on a ‘weak’ instance, a non-weak object is returned.
    #[inline]
    fn clone(&self) -> Self {
        Self::from_raw(unsafe { capi::pa_proplist_copy(self.0.ptr) })
    }
}
```

Reply:
Ah, good spot. I'd forgotten about and overlooked that. The fact that is actually already a proper mechanism for cloning `Proplist` and `Format::Info` objects helps a lot. I've now implemented `to_owned()` for all relevant structs.

Sorry for not ending up using your MR, but thanks for prompting me to finally get this done.